### PR TITLE
[CI]: Migrate CI tests to CircleCI from Travis.

### LINKED ISSUE #2641: [CI] CircleCI runs much faster than Travis. Migrate two last tests to the CircleCI.
Migrate last two CI tests to CircleCI from Travis:
* Clang format checks
* Coverage builds/reports

### PR DESCRIPTION
**Pull request by ZchiPitt**
Summary:
Migrate CI tests to CircleCI from Travis, which are
- Clang format checks
- Coverage builds/reports

[Optional Fixes #issue]
fixes #2641

### CODE REVIEW

**Comment by rdzhabarov** [comment]
Thank you for doing this work @ZchiPitt!

File: .circleci/build.sh [hunk 1]
```diff
@@ -70,6 +76,20 @@ elif [[ "$CIRCLE_JOB" == RELEASE_WITH_EXPENSIVE_TESTS ]]; then
...
+    sudo pip install awscli --upgrade
+    sudo apt-get install -qq g++-4.8
+    sudo update-alternatives --install /usr/bin/gcov gcov /usr/bin/gcov-4.8 90
+    CMAKE_ARGS=("CC=gcc-4.8 CXX=g++-4.8 
+          -DCMAKE_BUILD_TYPE=Debug -DGLOW_WITH_OPENCL=OFF -DGLOW_WITH_CPU=ON
+          -DLLVM_DIR=/usr/lib/llvm-7/cmake
+          -DCMAKE_CXX_FLAGS=-Werror
+          -DGLOW_USE_COVERAGE=ON")
+elif [[ "$CIRCLE_JOB" == CHECK_CLANG_FORMAT ]]; then
+    sudo apt-get install -y clang-format-8
```
Comment (rdzhabarov) [inline .circleci/build.sh]: make sure the right version is used so that clang-format produces the same result (as in travis/locally)

**Comment by rdzhabarov** [comment]
Were you able to compile successfully with gcc-4.8?
(If that works we can get rid of travis first and then follow up with the fix on gcov/gcc mismatch)

File: .circleci/build.sh [hunk 2]
```diff
@@ -24,24 +24,36 @@ install_pocl() {
...
-sudo ln -s /usr/bin/llvm-config-8 /usr/bin/llvm-config-8.0
-
-GLOW_DIR=$PWD
+    # Redirect clang
+    sudo ln -s /usr/bin/clang-8 /usr/bin/clang
+    sudo ln -s /usr/bin/clang++-8 /usr/bin/clang++
+    sudo ln -s /usr/bin/llvm-symbolizer-8 /usr/bin/llvm-symbolizer
+    sudo ln -s /usr/bin/llvm-config-8 /usr/bin/llvm-config-8.0
+else
+    sudo -E apt-add-repository -y "ppa:ubuntu-toolchain-r/test"
```
Comment (rdzhabarov) [inline .circleci/build.sh]: do we need to do all these things to install llvm-7? (that's probably just an artifact of this being run on travis and now we can get rid of this).
Just do similar things as you do above, but for llvm-7.

File: .circleci/run_coverage.sh [hunk 3]
```diff
@@ -1,20 +1,17 @@
 #!/bin/bash
-
-ninja all
-ninja glow_coverage
-
+cd build
```
Comment (rdzhabarov) [inline .circleci/run_coverage.sh]: why do we need to add this? (what has changed)

File: .circleci/test.sh [hunk 4]
```diff
@@ -58,7 +58,16 @@ case ${CIRCLE_JOB} in
     RELEASE_WITH_EXPENSIVE_TESTS)
         run_unit_tests check_expensive
         ;;
-
+    COVERAGE)
+        cd "${GLOW_SRC}"
+        .circleci/run_coverage.sh
+        ;;
+    CHECK_CLANG_FORMAT)
+        CLANG_COMMAND=/usr/bin/clang-format-7
```
Comment (rdzhabarov) [inline .circleci/test.sh]: where is it used?

File: .circleci/run_coverage.sh (hunk 3 above)
Comment (ZchiPitt) [reply .circleci/run_coverage.sh]: i've cleaned it

File: .circleci/test.sh (hunk 4 above)
Comment (ZchiPitt) [reply .circleci/test.sh]: inside the format.sh, there's variable called CLANG_COMMAND

**Comment by ZchiPitt** [comment]
Done with cleaning and all the two tests work successfully on CircleCI now.

File: .circleci/build.sh [hunk 5]
```diff
@@ -70,6 +76,18 @@ elif [[ "$CIRCLE_JOB" == RELEASE_WITH_EXPENSIVE_TESTS ]]; then
     CMAKE_ARGS+=("-DGLOW_MODELS_DIR=$MODELS_DIR")
     CMAKE_ARGS+=("-DGLOW_WITH_OPENCL=OFF")
     CMAKE_ARGS+=("-DCMAKE_BUILD_TYPE=Release")
+elif [[ "$CIRCLE_JOB" == "COVERAGE" ]]; then
+    sudo apt-get install wget 
+    sudo apt-get install -y lcov libgoogle-glog-dev
```
Comment (rdzhabarov) [inline .circleci/build.sh]: no need for libgoogle-glog-dev, please remove.

**Pull request closed without merging**